TESTS/host_tests/pyusb_basic.py

```diff
@@ -45,22 +51,189 @@ def get_interface(dev, interface, alternate=0):
+        self.log("Received product_id %s" % (product_id))
+
+        dev = self.find_device(serial_number)
+        if(dev == None):
+            return
+
+        try:
+            control_basic_test(dev, vendor_id, product_id, log=print)
+            self.report_success()
+        except (RuntimeError) as exc:
+            self.report_error(exc)
+
+
+    def _callback_control_stall_test(self, key, value, timestamp):
+        self.log("Received serial %s" % (value))
+
+        dev = self.find_device(value)
+        if(dev == None):
+            return
+
+        try:
+            control_stall_test(dev, log=print)
+            self.report_success()
+        except (RuntimeError) as exc:
+            self.report_error(exc)
+
+
+    def _callback_control_sizes_test(self, key, value, timestamp):
+        self.log("Received serial %s" % (value))
+
+        dev = self.find_device(value)
+        if(dev == None):
+            return
+
+        try:
+            control_sizes_test(dev, log=print)
+            self.report_success()
+        except (RuntimeError) as exc:
+            self.report_error(exc)
+
+
+    def _callback_control_stress_test(self, key, value, timestamp):
+        self.log("Received serial %s" % (value))
+
+        dev = self.find_device(value)
+        if(dev == None):
+            return
+
+        try:
+            control_stress_test(dev, log=print)
+            self.report_success()
+        except (RuntimeError) as exc:
+            self.report_error(exc)
+
+
+    def _callback_device_reset_test(self, key, value, timestamp):
+        self.log("Received serial %s" % (value))
+
+        dev = self.find_device(value)
+        if(dev == None):
+            return
+
+        try:
+            self.device_reset_test.send(dev)
+            self.report_success()
+        except (RuntimeError) as exc:
+            self.report_error(exc)
+
+    def _callback_device_soft_reconnection_test(self, key, value, timestamp):
+        self.log("Received serial %s" % (value))
+
+        dev = self.find_device(value)
+        if(dev == None):
+            return
+
+        try:
+            self.device_soft_reconnection_test.send(dev)
+            self.report_success()
+        except (RuntimeError) as exc:
+            self.report_error(exc)
+
+
+    def _callback_repeated_construction_destruction_test(self, key, value, timestamp):
+        self.log("Received serial %s" % (value))
+
+        dev = self.find_device(value)
+        if(dev == None):
+            return
+
+        try:
+            repeated_construction_destruction_test(dev, log=print)
+            self.report_success()
+        except (RuntimeError) as exc:
+            self.report_error(exc)
+
+    def find_device(self, serial_number):
+        dev = usb.core.find(custom_match=TestMatch(serial_number))
+        if dev is None:
+            self.log("Device not found")
+            self.send_kv("failed", "0")
+        return dev
```

Comment:
With the new reset tests I had to add a retry to get this to work correctly:
```
    def find_device(self, serial_number):
        for _ in range(20):
            dev = usb.core.find(custom_match=TestMatch(serial_number))
            if dev is not None:
                break
            time.sleep(0.1)
```

Reply:
done